src/jit/codegenarmarch.cpp

```diff
@@ -859,6 +859,154 @@ void CodeGen::genCodeForShift(GenTreePtr tree)
+
+    // Temp register used to perform the sequence of loads and stores.
+    regNumber tmpReg = cpObjNode->GetSingleTempReg();
+    assert(genIsValidIntReg(tmpReg));
+
+    unsigned slots = cpObjNode->gtSlots;
+    emitter* emit  = getEmitter();
+
+    BYTE* gcPtrs = cpObjNode->gtGcPtrs;
+
+    emitAttr attr;
+#if defined(_TARGET_ARM_)
+    attr = EA_4BYTE;
+#elif defined(_TARGET_ARM64_)
+    attr   = EA_8BYTE;
+#endif // _TARGET_ARM*
+
+    // If we can prove it's on the stack we don't need to use the write barrier.
+    if (dstOnStack)
+    {
+        // TODO-ARM64-CQ: Consider using LDP/STP to save codesize.
+        for (unsigned i = 0; i < slots; ++i)
+        {
+            if (gcPtrs[i] == GCT_GCREF)
+                attr = EA_GCREF;
+            else if (gcPtrs[i] == GCT_BYREF)
+                attr = EA_BYREF;
+#if defined(_TARGET_ARM_)
+            emit->emitIns_R_R_I(INS_ldr, attr, tmpReg, srcReg, TARGET_POINTER_SIZE);
+            emit->emitIns_R_R_I(INS_str, attr, tmpReg, dstReg, TARGET_POINTER_SIZE);
+#elif defined(_TARGET_ARM64_)
+            emit->emitIns_R_R_I(INS_ldr, attr, tmpReg, srcReg, TARGET_POINTER_SIZE, INS_OPTS_POST_INDEX);
+            emit->emitIns_R_R_I(INS_str, attr, tmpReg, dstReg, TARGET_POINTER_SIZE, INS_OPTS_POST_INDEX);
+#endif
+        }
+    }
+    else
+    {
+        unsigned gcPtrCount = cpObjNode->gtGcPtrCount;
+
+        unsigned i = 0;
+        while (i < slots)
+        {
+            switch (gcPtrs[i])
+            {
+                case TYPE_GC_NONE:
+// TODO-ARM64-CQ: Consider using LDP/STP to save codesize in case of contigous NON-GC slots.
+#if defined(_TARGET_ARM_)
+                    emit->emitIns_R_R_I(INS_ldr, attr, tmpReg, srcReg, TARGET_POINTER_SIZE);
+                    emit->emitIns_R_R_I(INS_str, attr, tmpReg, dstReg, TARGET_POINTER_SIZE);
+#elif defined(_TARGET_ARM64_)
+                    emit->emitIns_R_R_I(INS_ldr, attr, tmpReg, srcReg, TARGET_POINTER_SIZE, INS_OPTS_POST_INDEX);
+                    emit->emitIns_R_R_I(INS_str, attr, tmpReg, dstReg, TARGET_POINTER_SIZE, INS_OPTS_POST_INDEX);
+#endif
+                    break;
+
+                default:
+                    // In the case of a GC-Pointer we'll call the ByRef write barrier helper
+                    genEmitHelperCall(CORINFO_HELP_ASSIGN_BYREF, 0, EA_PTRSIZE);
+
+                    gcPtrCount--;
+                    break;
+            }
+            ++i;
+        }
+        assert(gcPtrCount == 0);
+    }
+
+    // Clear the gcInfo for registers of source and dest.
+    // While we normally update GC info prior to the last instruction that uses them,
+    // these actually live into the helper call.
+    regMaskTP sourceRegMask, destRegMask;
+#if defined(_TARGET_ARM_)
+    destRegMask   = RBM_ARG_0;
+    sourceRegMask = RBM_ARG_1;
+#elif defined(_TARGET_ARM64_)
+    destRegMask   = RBM_WRITE_BARRIER_DST_BYREF;
+    sourceRegMask = RBM_WRITE_BARRIER_SRC_BYREF;
```

Comment:
In general, I like using "full words". However, "src" and "dst" are widely used abbreviations, so I think it would be better in this function to use "src" instead of "source" and "dst" instead of "dest", consistently everywhere.